### PR TITLE
Commands for Create, List, Update, Delete, Stop, Attach and Remove Metrics Exporter Configs

### PR DESCRIPTION
**Pull request by adityakaushik99-yb**
- Adds CLI support for Metrics Exporter Project
- Tested all commands. They work as expected

### CODE REVIEW

**Review by posriniv — COMMENTED** [review]
Please remove internal issue numbers from the title since this is an open public repo.

**Comment by adityakaushik99-yb** [comment]
> Please review internal issue numbers from the title since this is an open public repo.

Sure, done

**Review by posriniv — APPROVED** [review]
Looks pretty good to me! Please make sure you run `make doc` before merging.

File: cmd/metrics_exporter/metrics_exporter.go [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+	Long:  "Create Metrics Exporter Config",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		metricsExporterName, _ := cmd.Flags().GetString("name")
+		metricsSinkType, _ := cmd.Flags().GetString("type")
+		datadogSpecString, _ := cmd.Flags().GetStringToString("datadog-spec")
+		apiKey := datadogSpecString["api-key"]
+		site := datadogSpecString["site"]
+
+		if strings.ToLower(metricsSinkType) == "datadog" {
```
Comment (cdavid) [inline cmd/metrics_exporter/metrics_exporter.go:53]: Please refactor a bit - the above DD related items should only be in the if branch.

File: cmd/metrics_exporter/metrics_exporter.go [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		metricsExporterName, _ := cmd.Flags().GetString("config-name")
+		metricsSinkType, _ := cmd.Flags().GetString("type")
+		datadogSpecString, _ := cmd.Flags().GetStringToString("datadog-spec")
```
Comment (cdavid) [inline cmd/metrics_exporter/metrics_exporter.go:279]: Validate that the flag was set before trying to use it.

File: cmd/metrics_exporter/metrics_exporter.go (hunk 1 above)
Comment (adityakaushik99-yb) [reply cmd/metrics_exporter/metrics_exporter.go:53]: They aren't actually DD specific, ApiKey and Site will be used in others as well

File: cmd/metrics_exporter/metrics_exporter.go [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+var createMetricsExporterCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create Metrics Exporter Config",
+	Long:  "Create Metrics Exporter Config",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		metricsExporterName, _ := cmd.Flags().GetString("name")
+		metricsSinkType, _ := cmd.Flags().GetString("type")
+		datadogSpecString, _ := cmd.Flags().GetStringToString("datadog-spec")
+		apiKey := datadogSpecString["api-key"]
```
[the remark this answers is not in the document]
Comment (shinji62) [reply cmd/metrics_exporter/metrics_exporter.go]: I refactor this part. please check @cdavid

File: cmd/metrics_exporter/metrics_exporter.go (hunk 2 above)
Comment (shinji62) [reply cmd/metrics_exporter/metrics_exporter.go]: Refactor this parts as well @cdavid

**Comment by shinji62** [comment]
@posriniv @cdavid I have updated the code and generate the docs.

**Review by shinji62 — APPROVED** [review]

**Pull request merged**